Allow board custom Makefiles to choose the name of the U-Boot's resulting binary
and the name it should take when copied to $(BINARIES_DIR).

Specific for Armadeus.

Signed-off-by: Julien Boibessot <julien.boibessot@armadeus.com>

Index: buildroot/boot/u-boot/u-boot.mk
===================================================================
--- buildroot.orig/boot/u-boot/u-boot.mk	2010-10-18 18:30:21.000000000 +0200
+++ buildroot/boot/u-boot/u-boot.mk	2010-10-18 18:43:14.000000000 +0200
@@ -34,10 +34,13 @@
 else
 U_BOOT_BIN:=u-boot.bin
 endif
+ifeq ($(U_BOOT_TARGET_BIN),)
+U_BOOT_TARGET_BIN:=u-boot-$(U_BOOT_VERSION)-$(DATE).bin
+endif
 
 MKIMAGE:=$(HOST_DIR)/usr/bin/mkimage
 
-U_BOOT_TARGETS:=$(BINARIES_DIR)/$(U_BOOT_BIN) $(MKIMAGE)
+U_BOOT_TARGETS:=$(BINARIES_DIR)/$(U_BOOT_TARGET_BIN) $(MKIMAGE)
 U_BOOT_ARCH=$(KERNEL_ARCH)
 
 # u-boot in the past used arch=ppc for powerpc
@@ -158,9 +161,9 @@
 		$(U_BOOT_MAKE_OPT) -C $(U_BOOT_DIR)
 
 # Copy the result to the images/ directory
-$(BINARIES_DIR)/$(U_BOOT_BIN): $(U_BOOT_DIR)/$(U_BOOT_BIN)
+$(BINARIES_DIR)/$(U_BOOT_TARGET_BIN): $(U_BOOT_DIR)/$(U_BOOT_BIN)
 	rm -f $(BINARIES_DIR)/$(U_BOOT_BIN)
-	cp -dpf $(U_BOOT_DIR)/$(U_BOOT_BIN) $(BINARIES_DIR)/
+	cp -dpf $(U_BOOT_DIR)/$(U_BOOT_BIN) $@
 
 ifneq ($(U_BOOT_COPYTO),)
 $(U_BOOT_COPYTO)/$(U_BOOT_TARGET_BIN): $(U_BOOT_DIR)/$(U_BOOT_BIN)
